agent/src/agent/api/routes/source.py

```diff
@@ -1,27 +1,50 @@
-from flask import jsonify, Blueprint
+from flask import jsonify, Blueprint, request
+from agent import source
+from agent.destination import HttpDestination
 from agent.repository import source_repository
 
-source = Blueprint('test', __name__)
+DESTINATION_DOESNT_EXIST = 'Destination is not configured. Please create agent destination first'
 
+sources = Blueprint('test', __name__)
 
-@source.route('/sources', methods=['GET'])
+
+@sources.route('/sources', methods=['GET'])
 def list_sources():
-    sources = []
-    for s in source_repository.get_all():
-        sources.append(s)
-    return jsonify(sources)
+    return jsonify(source_repository.get_all())
 
 
-@source.route('/sources', methods=['POST'])
+@sources.route('/sources', methods=['POST'])
 def create():
-    return 'Create a source'
-
-
-@source.route('/source/<source_id>', methods=['PUT'])
-def edit(source_id):
-    return 'Edit source'
-
-
-@source.route('/source/<source_id>', methods=['DELETE'])
+    if not HttpDestination.exists():
+        return DESTINATION_DOESNT_EXIST, 400
+    try:
+        source.validate_json_for_create(request.get_json())
+        source_instances = []
+        for config in request.get_json():
+            source_instances.append(source.create_from_json(config).to_dict())
+    except Exception as e:
+        return str(e), 400
```

Comment:
Generic exceptions should return server error 500. Returning 400 bad request all the time may be confusing because it might not be a client's fault but something broken inside the code.